File: src/_02_StructuralPattern/_02_04_Decorator/java/after/SpamFilteringCommentDecorator.java
package _02_StructuralPattern._02_04_Decorator.java.after;

public class SpamFilteringCommentDecorator extends CommentDecorator {
  
  public SpamFilteringCommentDecorator(CommentService commentService) {
    super(commentService);
  }

  @Override
  public void addComment(String comment) {
    if (!isNotSpam(comment)) return;

    super.addComment(comment);
  }
  
  private Boolean isNotSpam(String comment) {
    if(comment.contains("http")) return false;
    
    return true;
  }
  
}

File: src/_02_StructuralPattern/_02_04_Decorator/SpamFilteringCommentDecorator.js
import CommentDecorator from "./CommentDecorator.js";

/**
 * @typedef { import("./CommentService").default } CommentService
 */

export default class SpamFilteringCommentDecorator extends CommentDecorator {
  /** @param { CommentService } */
  constructor(commentService) {
    super(commentService);
  }

  /**
   * @override
   * @param { string } comment
   */
  addComment(comment) {
    if(!this.isNotSpam(comment)) return;

    super.addComment(comment);
  }
  
  /**
   * @param { string } comment
   * @returns { boolean }
   */
  isNotSpam(comment) {
    return comment.indexOf("http") < 0
  }
}